scripts/utils.js

```diff
@@ -46,4 +45,125 @@ searchBar.addEventListener('keyup', (event) => {
+
+            const heading = document.createElement("th");
+            heading.innerText = column.name;
+            heading.style.backgroundColor = "#6aa1b2";
+
+            //sortable: true --> if true make a sort button, align text to center to match button
+            if (column.sortable) {
+                const button = document.createElement("button");
+                button.innerText = "Sort";
+                heading.style.textAlign = "center";
+                heading.appendChild(button);
+            }
+            tableHeadRow.appendChild(heading);
+        }
+        table.appendChild(tableHeadRow)
+        
+    });
+}
+export {tableHeader};
+
+/*.................................................................................................................*/
+
+// Table Content creation
+var tableContent = (httpMethod, tableHeaderUrl, tableContentUrl) => {
+
+    createJsonRequest(httpMethod, tableHeaderUrl, function(err, tableColumns) {
+
+        let keys = [], types = [],texts =[];
+        for(let column of tableColumns) {
+            keys.push(column.key);
+            types.push(column.type);
+            texts.push(column.text);
+        }
+        
+        createJsonRequest(httpMethod, tableContentUrl, function(err, tableRows) {
+
+            const table = document.querySelector("#pageTable tbody");
+            console.log(tableRows)
+
+            for(let column of tableRows) {
+                const tableHeadRow = document.createElement("tr");
+
+                for (let key in column) {
+                    
+                    const tableData = document.createElement("td");
+                    const dataIndex = keys.indexOf(key);
+
+                    const isValid = (value) => {
+                        tableData.innerText = value != "-" ? value: "-";
+                    }
+
+                    const isLink = (link, value, text) => {
+                        if (value != "-") {
+                            link.style.textDecoration = "underline";
+                            link.setAttribute("href", value);
+                            link.innerText = text;
+                        }
+                        else {
+                            link.innerText = "-";
+                        }
+                        link.style.color = "#ffffff";
+                        tableData.appendChild(link);
+                    }
+
+                    const isAvailable = (button, availability) => {
+                        if (availability){
+                            tableData.appendChild(button);
+                        }
+                        else {
+                            tableData.innerText = "-";
+                            tableData.style.textAlign = "center";
+                        }
+                    }
+
+                    // valid column check
+                    if(keys.indexOf(key) != -1) {
+
+                        if (types[dataIndex] == "string"){
```

Comment:
"`types[dataIndex]`"  this expression can be assigned to a constant and the same can be used with the rest of the code instead of using the same expression

Reply:
Renamed the variables in the latest commit (logout branch)